bench.js

```diff
@@ -119,4 +120,107 @@ async function benchBranch(app, config) {
+        title: "Runtime Benchmarks Pallet",
+        branchCommand: 'cargo run --release --features runtime-benchmarks --manifest-path bin/node/cli/Cargo.toml -- benchmark --chain dev --steps 50 --repeat 20 --extrinsic "*" --raw --execution=wasm --wasm-execution=compiled --output --pallet',
+    },
+    "custom": {
+        title: "Runtime Benchmarks Custom",
+        branchCommand: 'cargo run --release --features runtime-benchmarks --manifest-path bin/node/cli/Cargo.toml -- benchmark',
+    }
+}
+
+function checkRuntimeBenchmarkCommand(command) {
+    let required = ["benchmark", "--pallet", "--extrinsic", "--execution", "--wasm-execution", "--steps", "--repeat", "--chain"];
+    let missing = [];
+    for (const flag of required) {
+        if (!command.includes(flag)) {
+            missing.push(flag);
+        }
+    }
+
+    return missing;
+}
+
+async function benchmarkRuntime(app, config) {
+    app.log("Waiting our turn to run benchmark...")
+
+    const release = await mutex.acquire();
+
+    try {
+        var benchConfig = RuntimeBenchmarkConfigs[config.id || "pallet"];
+        var extra = config.extra || "";
+
+        if (extra.length == 0) {
+            return errorResult(`Incomplete command.`)
+        }
+
+        // Append extra flags to the end of the command
+        benchConfig.branchCommand += " " + extra;
+
+        let missing = checkRuntimeBenchmarkCommand(benchConfig.branchCommand);
+        let output = benchConfig.branchCommand.includes("--output");
+
+        if (missing.length > 0) {
+            return errorResult(`Missing required flags: ${missing.toString()}`)
+        }
+
+        var benchContext = new BenchContext(app, config);
+        console.log(`Started runtime benchmark "${benchConfig.title}."`);
+        shell.mkdir("git")
+        shell.cd(cwd + "/git")
+
+        var { error } = benchContext.runTask(`git clone ${config.repository}`, "Cloning git repository...");
+        if (error) {
+            app.log("Git clone failed, probably directory exists...");
+        }
+
+        shell.cd(cwd + "/git/substrate");
+
+        var { error, stderr } = benchContext.runTask(`git fetch`, "Doing git fetch...");
+        if (error) return errorResult(stderr);
+
+        // Checkout the custom branch
+        var { error, stderr } = benchContext.runTask(`git checkout ${config.branch}`, `Checking out ${config.branch}...`);
+        if (error) {
+            app.log("Git checkout failed, probably some dirt in directory... Will continue with git reset.");
+        }
+
+        var { error, stderr } = benchContext.runTask(`git reset --hard origin/${config.branch}`, `Resetting ${config.branch} hard...`);
+        if (error) return errorResult(stderr);
+
+        benchConfig.preparationCommand && benchContext.runTask(benchConfig.preparationCommand);
+
+        // Merge master branch
+        var { error, stderr } = benchContext.runTask(`git merge origin/${config.baseBranch}`, `Merging branch ${config.baseBranch}`);
```

Comment:
Do you need to merge it?

in the performance benchmarks, it is used for comparison, here I'm not sure

Reply:
i guess i would just want to merge in order to make sure we are running the benchmarks on the potential "latest changes". I mean nothing will strictly enforce that these benchmarks always happen on the latest substrate, but this will at least check when benchmarks are running

Reply:
if you think it doesnt make sense, i have no strong opinion to keep it

Reply:
Looks right, this merge will be pushed with the benchmarking result btw

Reply:
good point, I should probably split the two